modules/coursier/shared/src/main/scala/coursier/Resolve.scala

```diff
@@ -180,7 +180,7 @@ final class Resolve[F[_]] private[coursier] (private val params: Resolve.Params[
           }
       }
 
-    S.bind(S.bind(run(initialRes))(S.point)) { res0 =>
```

Comment:
I think I needed this because initialRes already contains resolution conflict, and validate0 would throw there before my rule can get to it.

Reply:
I think I just need to move `validate0` to somewhere so it gets run at least once.